packages/core/src/components/fields/BooleanField.tsx

```diff
@@ -52,14 +53,21 @@ function BooleanField<T = any, F = any>(props: FieldProps<T, F>) {
         .filter((o) => o) as RJSFSchemaDefinition[], // cast away the error that typescript can't grok is fixed
     });
   } else {
-    enumOptions = optionsList({
-      enum: schema.enum || [true, false],
-      enumNames:
-        schema.enumNames ||
-        (schema.enum && schema.enum[0] === false
-          ? ["No", "Yes"]
-          : ["Yes", "No"]),
-    });
+    schema.enum = schema.enum ?? [true, false];
+    if (
+      schema.enum &&
+      schema.enum.length === 2 &&
+      schema.enum.every((v) => typeof v === "boolean")
+    ) {
+      enumOptions = [
+        { value: schema.enum[0], label: schema.enum[0] ? "Yes" : "No" },
+        { value: schema.enum[1], label: schema.enum[1] ? "Yes" : "No" },
+      ];
```

Comment:
@heath-freenome instead of changing the snapshots, I opted to maintain the existing behavior of labeling boolean enums as "Yes"/"No".

Reply:
Keep this as it is, don't use `enumNames`

Reply:
I think we still need to check for user-supplied enumNames, but we can continue to not use enumNames for the `[true, false]` case